### DISCUSSION TITLE
UART1 pin init in ESP32-C3

### DISCUSSION THREAD

**Discussion by pablogventura** [micropython/micropython#12795] [ESP32]
Hi! I have a MicroPython code that is runing in MicroPython 1.20 in a ESP32-C3 using this line to set the pins to use UART1:
`machine.UART(1, baudrate=115200, tx=18, rx=19)`

I was trying to update to MicroPython 1.21 and I noticed that now that line raises this excepction:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid pin
```

I am doing something wrong? Can be possible it is a bug?

Thank you in advance.

**Comment by puppet13th** [comment]
from version 1.21 they merged build for ESP32_GENERIC_C3, earlier version have 2 build with and without USB.
i just compiled C3 firmware without usb function, and can use pin 18 and 19.
remove this [line](https://github.com/micropython/micropython/blob/95d8b5fd555ba26ebd79b29fade45d2bf0abc6c4/ports/esp32/boards/ESP32_GENERIC_C3/mpconfigboard.cmake#L6) before compiling the firmware.

**Reply by Seruf-Che** [reply]
Hi,
Could you please provide a link to the compiled version of firmware?

**Comment by pablogventura** [comment]
It's works perfect! Thanks you a lot!!!
